src/tests/functional/link_prefetch_observer_tests.js

```diff
@@ -296,6 +293,13 @@ const assertNotPrefetchedOnHover = async ({ page, selector, callback }) => {
   assert.equal(requestMade, false, "Network request was made when it should not have been.")
 }
 
+const assertRequestNotMadeOnClick = async ({ page, selector }) => {
+  let requestMade = false
+  page.on("request", async (request) => (requestMade = true))
+  await clickSelector({ page, selector })
+  assertRequestNotMade(requestMade)
```

Comment:
Can we add a callback to the `assertRequestMade` / `assertRequestNotMade` helpers so we can move the `requestMade` and event handler declaration to the helpers, so this looks like:

```js
assertRequestNotMade(async () => {
  await clickSelector({ page, selector })
})
```

Reply:
Sorry, just got free right now to tackle this.

Good call, love the resulting abstraction. LMK if this is what you had in mind.